test/new-e2e/tests/npm/agentenv_npm_test.go

```diff
@@ -27,18 +27,19 @@ func (v *vmSuiteEx6) Test1_FakeIntakeNPM() {
 
 	// force pulumi to deploy before running the test
 	v.Env().VM.Execute("curl http://httpbin.org/anything")
+	v.Env().Fakeintake.FlushServerAndResetAggregators()
 
 	// This loop waits for agent and system-probe to be ready, stated by
 	// checking we eventually receive a payload
 	v.EventuallyWithT(func(c *assert.CollectT) {
 		v.Env().VM.Execute("curl http://httpbin.org/anything")
 
 		hostnameNetID, err := v.Env().Fakeintake.GetConnectionsNames()
-		assert.NoError(c, err, "fakeintake GetConnectionsNames() error")
-
-		assert.NotZero(c, len(hostnameNetID), "no connections yet")
+		if !assert.NoError(c, err, "fakeintake GetConnectionsNames() error") {
+			return
+		}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
`require` panics on not respected assertion, so would act similarly - `panics` are recovered by the `TestSuite` normally
```suggestion
		require.NoError(c, err, "fakeintake GetConnectionsNames() error")
```

Reply:
I didn't see this behavior as require will call c.FailNow() so this will stop the current v.EventuallyWithT and make the test failed, so not recovered

```
panic: Assertion failed

goroutine 296 [running]:
github.com/stretchr/testify/assert.(*CollectT).FailNow(0x4182f00?)
	/home/nplanel/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1874 +0x27
github.com/stretchr/testify/require.NoError({0x41b1a00, 0xc000531590}, {0x417b960, 0xc000398f60}, {0xc000aa7f88, 0x1, 0x1})
	/home/nplanel/go/pkg/mod/github.com/stretchr/testify@v1.8.4/require/require.go:1360 +0xb4
github.com/DataDog/datadog-agent/test/new-e2e/tests/npm.(*vmSuiteEx6).Test1_FakeIntakeNPM.func1(0x0?)
	/home/nplanel/dd/datadog-agent/test/new-e2e/tests/npm/agentenv_npm_test.go:39 +0x105
github.com/stretchr/testify/assert.EventuallyWithT.func1()
	/home/nplanel/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1933 +0x2f
created by github.com/stretchr/testify/assert.EventuallyWithT
	/home/nplanel/go/pkg/mod/github.com/stretchr/testify@v1.8.4/assert/assertions.go:1932 +0x28a
FAIL	github.com/DataDog/datadog-agent/test/new-e2e/tests/npm	165.070s
```